src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -795,6 +795,7 @@ def sql_query(self) -> str:
             placeholder = f"{analyzer_utils.LEFT_PARENTHESIS}{self.from_._id}{analyzer_utils.RIGHT_PARENTHESIS}"
             self._sql_query = self.placeholder_query.replace(placeholder, from_clause)
         else:
+            self.analyzer.subquery_plans = []
```

Comment:
this is probably going to be problematic also, because this part can potentially be done during the middle of resolving, and if we clears the subquery_plans here, we might cause problem to the plan node that is actually for resolving

Reply:
you are right. I missed this possibility. I can read current value before resetting it and calling analyze.